### PR TITLE
More cmd flags and delay to prevent rate limits

### PR DESCRIPTION
**Pull request by xanonid**
Closes: #4

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -2,35 +2,68 @@
...
-		UsageText: "terraform-state-mover [-- <terraform args>]",
+		Name:    "terraform-state-mover",
+		Usage:   "refactoring Terraform code has never been easier",
+		Authors: []*cli.Author{{Name: "Maximilian Bode", Email: "maxbode@gmail.com"}},
+		Action:  action,
+		Flags: []cli.Flag{
+			&cli.DurationFlag{
+				Name: "delay", Aliases: []string{"d"},
+				Usage: "Delay between terraform state mv calls. Helps to avoid rate-limits.",
+				Value: time.Second * 2,
```
Comment (mbode) [inline main.go:40]: I'm inclined to use a default of zero seconds so as users with providers without rate limiting are not slowed down unnecessarily. What do you think?

File: terraform.go [hunk 2]
```diff
@@ -40,8 +42,15 @@ type Resource struct {
...
 }
 
-func terraformExec(args []string, extraArgs ...string) error {
+func terraformExec(cfg config, executeInDryRun bool, args []string, extraArgs ...string) error {
 	args = append(extraArgs, args...)
+	if cfg.dryrun && !executeInDryRun {
+		fmt.Println("Dry-run - would have called now terraform", strings.Join(args, " "))
+		return nil
+	}
+	if cfg.verbose {
```
[the remark this answers is not in the document]
Comment (xanonid) [reply terraform.go]: yes - fixed

File: planner.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ func changes(args []string) ([]ResChange, error) {
 	tfPlanName := tfPlan.Name()
 	defer os.Remove(tfPlanName)
 
-	if err := terraformExec(args, "plan", "-out="+tfPlanName); err != nil {
+	if err := terraformExec(config{}, true, args, "plan", "-out="+tfPlanName); err != nil {
```
[the remark this answers is not in the document]
Comment (xanonid) [reply planner.go]: also fixed.

File: main.go (hunk 1 above)
Comment (xanonid) [reply main.go]: makes sense. Changed the default to "0s" (i.e. no delay). I added a hint to the README.md for GCP provider user where the delay flag is often necessary.

**Pull request merged**